File: Python/330A-cakeminator.py
r, c = map(int, input().split())
cakecells = []
for i in range(r):
    row = input()
    cakecells.append(row)
rowstoeat = 0
for item in cakecells:
    rowstoeat += 1 if 'S' not in item else 0
columnstoeat = 0
for j in range(c):
    eatable = True
    for i in range(r):
        if cakecells[i][j] == 'S':
            eatable = False
            break
    columnstoeat += 1 if eatable else 0
print((rowstoeat * c) + (columnstoeat * (r - rowstoeat)))

File: C++/A/330A-cakeminator.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int r, c, answer;
    cin >> r >> c;
    vector<string> cakecells(r);
    for (int i = 0; i < r; i++)
        cin >> cakecells[i];
    int rowstoeat = 0, columnstoeat = 0;
    for (string item : cakecells)
        (item.find('S') == string::npos) ? rowstoeat++ : 0;
    for (int j = 0; j < c; j++)
    {
        bool eatable = true;
        for (int i = 0; i < r; i++)
            if (cakecells[i][j] == 'S')
            {
                eatable = false;
                break;
            }
        columnstoeat += ((eatable) ? 1 : 0);
    }
    cout << (rowstoeat * c) + (columnstoeat * (r - rowstoeat));
}